src/router/middlewares/setServerUrl.ts

```diff
@@ -0,0 +1,27 @@
+/* eslint-disable @typescript-eslint/no-unused-vars */
+/*
+ *  Copyright Amazon.com, Inc. or its affiliates. All Rights Reserved.
+ *  SPDX-License-Identifier: Apache-2.0
+ *
+ */
+
+import { FhirConfig } from 'fhir-works-on-aws-interface';
+import express from 'express';
+import RouteHelper from '../routes/routeHelper';
+
+/**
+ * Sets the value of `res.locals.serverUrl`
+ * the serverUrl can either be a static value from FhirConfig of a dynamic value for some multi-tenancy setups.
+ */
+export const setServerUrlMiddleware: (
+    fhirConfig: FhirConfig,
+) => (req: express.Request, res: express.Response, next: express.NextFunction) => void = (fhirConfig: FhirConfig) => {
+    return RouteHelper.wrapAsync(async (req: express.Request, res: express.Response, next: express.NextFunction) => {
+        if (req.baseUrl && req.baseUrl !== '/') {
+            res.locals.serverUrl = fhirConfig.server.url + req.baseUrl;
```

Comment:
So `fhirConfig.server.url` is going to be the API_URL environment variable (`apigw.com/dev`) and `req.baseUrl` is the FHIR side (`/Patient/1234`)

Where would the tenant portion reside?

Reply:
> req.baseUrl is the FHIR side (/Patient/1234)

That's incorrect, `/Patient/1234` would be `req.path`. `req.baseUrl` is the the URL path on which the router instance was mounted.

In our case the main router is either mounte on `/` or `tenant/:tenantId` depending in the value of `useTenantSpecificUrl`